ceno_zkvm/src/circuit_builder.rs

```diff
@@ -334,13 +334,15 @@ impl<E: ExtensionField> ConstraintSystem<E> {
     pub fn r_table_record<NR, N>(
         &mut self,
         name_fn: N,
+        ram_type: RAMType,
         table_spec: SetTableSpec,
-        rlc_record: Expression<E>,
+        record: Vec<Expression<E>>,
     ) -> Result<(), ZKVMError>
     where
         NR: Into<String>,
         N: FnOnce() -> NR,
     {
+        let rlc_record = self.rlc_chip_record(record.clone());
```

Comment:
Because right now we have urgent requirement to debug in example, I think some workaround is understand

We can make those debug information toggle on/off by a specific feature

```diff
diff --git a/ceno_zkvm/src/circuit_builder.rs b/ceno_zkvm/src/circuit_builder.rs
index 92682a67..316fa4f9 100644
--- a/ceno_zkvm/src/circuit_builder.rs
+++ b/ceno_zkvm/src/circuit_builder.rs
@@ -386,7 +386,7 @@ impl<E: ExtensionField> ConstraintSystem<E> {
     pub fn read_record<NR: Into<String>, N: FnOnce() -> NR>(
         &mut self,
         name_fn: N,
-        ram_type: RAMType,
+        #[cfg(feature = "debug_lookup_failed")] ram_type: RAMType,
         record: Vec<Expression<E>>,
     ) -> Result<(), ZKVMError> {
         let rlc_record = self.rlc_chip_record(record.clone());
@@ -399,9 +399,12 @@ impl<E: ExtensionField> ConstraintSystem<E> {
         self.r_expressions.push(rlc_record);
         let path = self.ns.compute_path(name_fn().into());
         self.r_expressions_namespace_map.push(path);
-        // Since r_expression is RLC(record) and when we're debugging
-        // it's helpful to recover the value of record itself.
-        self.r_ram_types.push((ram_type, record));
+        #[cfg(feature = "debug_lookup_failed")]
+        {
+            // Since r_expression is RLC(record) and when we're debugging
+            // it's helpful to recover the value of record itself.
+            self.r_ram_types.push((ram_type, record));
+        }
         Ok(())
     }
```

Reply:
No no I can live with that, I’ll suggest some name or documentation later.